### PR TITLE
Create homepage layout

### LINKED ISSUE #3: Create Homepage Layout
**Task**:
- Develop the layout for the homepage, including sections for featured products, categories, etc.
- Ensure the layout is responsive and looks good on different screen sizes.

**Acceptance Criteria**:
- Homepage layout is designed and implemented.
- Sections for featured products, categories, etc., are added.
- Layout is responsive and looks good on different devices.

### PR DESCRIPTION
**Pull request by FrancoisLunick**
This pull request implements the initial layout for the homepage of the Amazon Clone project. It includes the following components and styles:

### Components

- **NavigationBar**: The navigation bar at the top of the page with links to the homepage, login, signup, and cart.
- **FeaturedProducts**: A section displaying featured products.
- **Categories**: A section displaying product categories.

### Styles

- **NavigationBar.css**: Styles for the navigation bar, including layout, spacing, and color settings.
- **FeaturedProducts.css**: Styles for the featured products section, including grid layout and individual product styling.
- **Categories.css**: Styles for the categories section, including grid layout and individual category styling.
- **Homepage.css**: General styles for the homepage layout, including responsive styles for different screen sizes.

### Changes

- `src/components/NavigationBar.js`: Added NavigationBar component with links and styles.
- `src/components/FeaturedProducts.js`: Added FeaturedProducts component with placeholder products.
- `src/components/Categories.js`: Added Categories component with placeholder categories.
- `src/styles/NavigationBar.css`: Added styles for NavigationBar component.
- `src/styles/FeaturedProducts.css`: Added styles for FeaturedProducts component.
- `src/styles/Categories.css`: Added styles for Categories component.
- `src/styles/Homepage.css`: Added styles for Homepage component, including responsive design.
- `src/Homepage.js`: Updated Homepage component to include NavigationBar, FeaturedProducts, and Categories components.

### Acceptance Criteria

- Homepage layout is designed and implemented.
- Sections for featured products, categories, etc., are added.
- Layout is responsive and looks good on different devices.

### Testing

- Verified that the homepage displays correctly with all sections included.
- Ensured that the layout is responsive and adapts to different screen sizes.

### CODE REVIEW

File: amazon_clone/src/Homepage.js [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
 
+/**
+ * Renders the Homepage component.
+ * 
+ * @returns {JSX.Element} The Homepage component 
+ */
 export default function Homepage() {
   return (
-    <div>Homepage</div>
+    <div className='homepage'>Homepage
```
Comment (Cybersunami) [inline amazon_clone/src/Homepage.js]: "Homepage" should not be visible.

**Comment by Cybersunami** [comment]
In App.js the root path is still the signup page.

**Review by Cybersunami — CHANGES_REQUESTED** [review]
The Homepage text is visible and the root path is set to the signup page.
Screenshot 2024-06-30 at 6 35 18 PM (2)

File: amazon_clone/src/Homepage.js (hunk 1 above)
Comment (FrancoisLunick) [reply amazon_clone/src/Homepage.js]: I removed the text

**Review by Cybersunami — APPROVED** [review]

**Pull request merged**